dashboard/pages/fragment_graph.tsx

```diff
@@ -210,9 +211,47 @@ export default function Streaming() {
         }
       }
     }
-    return () => {}
+    return () => { }
   }, [relationId, relationList, setRelationId])
 
+
+  // get back pressure rate without prometheus
+  const [backPressuresMetricsWithoutPromtheus, setBackPressuresMetrics] =
+    useState<BackPressuresMetrics>()
+  const [previousBP, setPreviousBP] = useState<BackPressureInfo[]>([])
+  const [backPressureRate, setBackPressureRate] = useState<
+    BackPressureRateInfo[]
+  >([])
+  const toast = useErrorToast()
+
+  useEffect(() => {
+    let localPreviousBP = previousBP
+
+    async function doFetch() {
+      while (true) {
+        try {
+          const currentBP = await getBackPressureWithoutPrometheus()
+          const metrics = calculateBPRate(currentBP, localPreviousBP)
+          setBackPressureRate(backPressureRate)
+          localPreviousBP = currentBP
+          setPreviousBP(currentBP)
+          metrics.outputBufferBlockingDuration = sortBy(
+            metrics.outputBufferBlockingDuration,
+            (m) => (m.metric.fragment_id, m.metric.downstream_fragment_id)
+          )
+
+          setBackPressuresMetrics(metrics)
+          await new Promise((resolve) => setTimeout(resolve, INTERVAL))
+        } catch (e: any) {
+          toast(e, "warning")
+          break
+        }
+      }
+    }
+    doFetch()
+    return () => { }
+  }, [toast])
```

Comment:
The second argument of `useEffect` - `dependencies`, tells React to re-run the function once any value of it is changed. Here you pass a `toast`? Seems to be a misunderstanding

Reply:
I follow the code style in the codebase.
https://github.com/risingwavelabs/risingwave/blob/4500a085264626db95ee13bd730870dede0d258b/dashboard/components/BackPressureTable.tsx#L59

https://github.com/risingwavelabs/risingwave/blob/4500a085264626db95ee13bd730870dede0d258b/dashboard/pages/cluster.tsx#L173

I know it will trigger re-run. But in current logic, no parameters suitable for here. So I just put the `toast`.

Reply:
Oh, do some more investigation. Maybe this is the problem why there is a mismatch between my impelmentation and the previous impelementation. Will remove the `toast`.

Reply:
You can run `npm run lint` and the linter will help to fix that.

Reply:
https://github.com/risingwavelabs/risingwave/compare/eric/no_toast 

I tried removing it from `dependencies` and everything seemed to work. cc. @BugenZhao Can you also take a try?

Reply:
> You can run npm run lint and the linter will help to fix that. Generally all outside variables captured by the closure should be included.

I tried `npm run lint`, and it will remind me to add the dependencies `backPressureRate` and `previousBP`. I applied the suggestion, but it will cause the function do infinite quickly re-trigger which is not we want here. 
So in current logic, the `backPressureRate` and `previousBP` are not suitable to put into dependencies. 
If you do think we should follow the lint guidence. I may need to figure out a way refactor the code logic.